### ISSUE TITLE
Disabling tests seems to cause an endless loop if a disabled test codeunit is added to multiple suites

### ISSUE THREAD

**Issue by tfenster** [microsoft/navcontainerhelper#719]
Hello,

if I am not misreading the output, it seems to me like Run-TestsInNavContainer runs into an endless loop if you try to disable a test codeunit which is referenced in multiple test suites. Here is my repro:

I am using the following script to create a new test container

```
[CmdletBinding()]
param (
    [Parameter()]
    [string]
    $ContainerName,
    
    [Parameter()]
    [string]
    $ImageName,
    
    [Parameter()]
    [string]
    $LicenseFile,
    
    [Parameter(Mandatory=$false)]
    [pscredential]
    $Credential = (New-Object System.Management.Automation.PSCredential -ArgumentList ("admin", (ConvertTo-SecureString -String "5uperSecret!" -AsPlainText -Force))),
    
    [Parameter(Mandatory=$false)]
    [string]
    $XunitResultsBasePath = "C:\ProgramData\NavContainerHelper\Extensions\$ContainerName"
)

$xunitResultsFile = Join-Path $XunitResultsBasePath "results-{0}.xml"

if ($null -ne (Get-BCContainers | Where-Object { $_ -eq $ContainerName })) {
    Write-Host "Remove existing container"
    Remove-BCContainer $ContainerName
}

Write-Host "Create new container"
New-BCContainer `
    -accept_eula `
    -accept_outdated `
    -containerName $ContainerName `
    -imageName $ImageName `
    -alwaysPull `
    -licenseFile $LicenseFile `
    -Credential $Credential `
    -includeTestToolkit `
    -updateHosts `
    -additionalParameters @("--env CustomWebSettings=SessionTimeout=01:00:00", "--env CustomNavSettings=EnableTaskScheduler=False") #`
    #-assignPremiumPlan 

docker cp $ContainerName:C:/Applications/TestFramework/TestRunner/DisabledTests.json .
$disabledTests = Get-Content "DisabledTests.json" | ConvertFrom-Json
$myDisabledTests = Get-Content "MyDisabledTests.json" | ConvertFrom-Json
$disabledTests += $myDisabledTests

Write-Host "Start tests"    
Get-BCContainerAppInfo $ContainerName | Where-Object { $_.Name.StartsWith("Tests-") } | ForEach-Object { 
    Write-Host ("Run tests for extension {0}" -f $_.Name)
    Run-TestsInBCContainer `
        -containerName $ContainerName `
        -credential $Credential `
        -XUnitResultFileName ($xunitResultsFile -f $_.Name.Substring(6)) `
        -extensionId $_.AppId `
        -disabledTests $disabledTests #`
        #-detailed `
        #-debugMode
}
```

My own disabled tests are stored in MyDisabledTests.json which looks like this

```
[
    {
        "codeunitName": "ERM Cash Flow Filling I",
        "method" : "*"
    }
]
```

I call this e.g. with the following parameters:

```
.\RunAllTests.ps1 -ContainerName autotestme -ImageName "mcr.microsoft.com/businesscentral/onprem:1910-w1-ltsc2019" -LicenseFile "C:\Users\Verwalter\Desktop\autotest\2019-10 1910 DE.flf"
```

The output looks like this:

```
.\RunAllTests.ps1 -ContainerName autotestme -ImageName "mcr.microsoft.com/businesscentral/onprem:1910-w1-ltsc2019" -LicenseFile "C:\Users\Verwalter\Desktop\autotest\2019-10 1910 DE.flf"
Create new container
NavContainerHelper is version 0.6.4.16
NavContainerHelper is running as administrator
Host is Microsoft Windows Server 2019 Datacenter - ltsc2019
Docker Client Version is 19.03.2
Docker Server Version is 19.03.2
Pulling image mcr.microsoft.com/businesscentral/onprem:1910-w1-ltsc2019
1910-w1-ltsc2019: Pulling from businesscentral/onprem
Digest: sha256:243e509934ae8fdcb0566776823966fd5a91e6a87b0336a24f334cd16cbee2b2
Using image mcr.microsoft.com/businesscentral/onprem:1910-w1-ltsc2019
Creating Container autotestme
Version: 15.0.36560.0-W1
Style: onprem
Platform: 15.0.36510.0
Generic Tag: 0.0.9.95
Container OS Version: 10.0.17763.737 (ltsc2019)
Host OS Version: 10.0.17763.805 (ltsc2019)
WARNING: The container operating system matches the host operating system, but the revision is different.
If you encounter issues, you might want to specify -isolation hyperv
Using locale en-US
Using process isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\Users\Verwalter\Desktop\autotest\2019-10 1910 DE.flf
Files in C:\ProgramData\NavContainerHelper\Extensions\autotestme\my:
- AdditionalOutput.ps1
- license.flf
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container autotestme from image mcr.microsoft.com/businesscentral/onprem:1910-w1-ltsc2019
67734cf43c90f5c0f6d88d17760a1782d099526094ce169d92062292b60c28c8
Waiting for container autotestme to be ready
Initializing...
Setting host.containerhelper.internal to 172.17.240.1 in container hosts file
Starting Container
Hostname is
PublicDnsName is autotestme
Using Windows Authentication
Starting Local SQL Server
Starting Internet Information Server
Modifying Service Tier Config File with Instance Specific Settings
Modifying Service Tier Config File with settings from environment variable
Setting EnableTaskScheduler to False
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Modifying Web Client config with settings from environment variable
Setting SessionTimeout to 01:00:00
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Creating Windows user admin
Setting SA Password and enabling SA
Creating SUPER user
Container IP Address: 172.17.243.191
Container Hostname  :
Container Dns Name  : autotestme
Web Client          : http://autotestme/BC/
Dev. Server         : http://autotestme
Dev. ServerInstance : BC
Setting autotestme to 172.17.243.191 in host hosts file

Files:
http://autotestme:8080/al-4.0.176004.vsix

Initialization took 65 seconds
Ready for connections!
Reading CustomSettings.config from autotestme
Creating Desktop Shortcuts for autotestme
Copy from container autotestme (C:\Applications\TestFramework\TestLibraries\Any\Microsoft_Any.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Any.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Any.app
Synchronizing Any on tenant default
Installing Any on tenant default
App successfully published
Copy from container autotestme (C:\Applications\TestFramework\TestLibraries\Assert\Microsoft_Library Assert.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Library Assert.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Library Assert.app
Synchronizing Library Assert on tenant default
Installing Library Assert on tenant default
App successfully published
Copy from container autotestme (C:\Applications\TestFramework\TestRunner\Microsoft_Test Runner.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Test Runner.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Test Runner.app
Synchronizing Test Runner on tenant default
Installing Test Runner on tenant default
App successfully published
Copy from container autotestme (C:\Applications\System Application\Test\Microsoft_System Application Test Library.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_System Application Test Library.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_System Application Test Library.app
Synchronizing System Application Test Library on tenant default
Installing System Application Test Library on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-TestLibraries.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-TestLibraries.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-TestLibraries.app
Synchronizing Tests-TestLibraries on tenant default
Installing Tests-TestLibraries on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Bank.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Bank.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Bank.app
Synchronizing Tests-Bank on tenant default
Installing Tests-Bank on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Cash Flow.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Cash Flow.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Cash Flow.app
Synchronizing Tests-Cash Flow on tenant default
Installing Tests-Cash Flow on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Cost Accounting.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Cost Accounting.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Cost Accounting.app
Synchronizing Tests-Cost Accounting on tenant default
Installing Tests-Cost Accounting on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-CRM integration.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-CRM integration.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-CRM integration.app
Synchronizing Tests-CRM integration on tenant default
Installing Tests-CRM integration on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Data Exchange.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Data Exchange.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Data Exchange.app
Synchronizing Tests-Data Exchange on tenant default
Installing Tests-Data Exchange on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Dimension.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Dimension.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Dimension.app
Synchronizing Tests-Dimension on tenant default
Installing Tests-Dimension on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-ERM.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-ERM.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-ERM.app
Synchronizing Tests-ERM on tenant default
Installing Tests-ERM on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Fixed Asset.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Fixed Asset.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Fixed Asset.app
Synchronizing Tests-Fixed Asset on tenant default
Installing Tests-Fixed Asset on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-General Journal.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-General Journal.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-General Journal.app
Synchronizing Tests-General Journal on tenant default
Installing Tests-General Journal on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Graph.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Graph.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Graph.app
Synchronizing Tests-Graph on tenant default
Installing Tests-Graph on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Integration.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Integration.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Integration.app
Synchronizing Tests-Integration on tenant default
Installing Tests-Integration on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Invoicing.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Invoicing.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Invoicing.app
Synchronizing Tests-Invoicing on tenant default
Installing Tests-Invoicing on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Job.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Job.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Job.app
Synchronizing Tests-Job on tenant default
Installing Tests-Job on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Local.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Local.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Local.app
Synchronizing Tests-Local on tenant default
Installing Tests-Local on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Misc.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Misc.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Misc.app
Synchronizing Tests-Misc on tenant default
Installing Tests-Misc on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Permissions.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Permissions.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Permissions.app
Synchronizing Tests-Permissions on tenant default
Installing Tests-Permissions on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Physical Inventory.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Physical Inventory.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Physical Inventory.app
Synchronizing Tests-Physical Inventory on tenant default
Installing Tests-Physical Inventory on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Prepayment.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Prepayment.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Prepayment.app
Synchronizing Tests-Prepayment on tenant default
Installing Tests-Prepayment on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Rapid Start.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Rapid Start.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Rapid Start.app
Synchronizing Tests-Rapid Start on tenant default
Installing Tests-Rapid Start on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Report.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Report.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Report.app
Synchronizing Tests-Report on tenant default
Installing Tests-Report on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Resource.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Resource.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Resource.app
Synchronizing Tests-Resource on tenant default
Installing Tests-Resource on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Reverse.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Reverse.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Reverse.app
Synchronizing Tests-Reverse on tenant default
Installing Tests-Reverse on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-SCM.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-SCM.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-SCM.app
Synchronizing Tests-SCM on tenant default
Installing Tests-SCM on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-SMB.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-SMB.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-SMB.app
Synchronizing Tests-SMB on tenant default
Installing Tests-SMB on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-SMTP.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-SMTP.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-SMTP.app
Synchronizing Tests-SMTP on tenant default
Installing Tests-SMTP on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Upgrade.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Upgrade.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Upgrade.app
Synchronizing Tests-Upgrade on tenant default
Installing Tests-Upgrade on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-User.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-User.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-User.app
Synchronizing Tests-User on tenant default
Installing Tests-User on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-VAT.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-VAT.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-VAT.app
Synchronizing Tests-VAT on tenant default
Installing Tests-VAT on tenant default
App successfully published
Copy from container autotestme (C:\Applications\BaseApp\Test\Microsoft_Tests-Workflow.app) to C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Workflow.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\autotestme\_Microsoft_Tests-Workflow.app
Synchronizing Tests-Workflow on tenant default
Installing Tests-Workflow on tenant default
App successfully published
TestToolkit successfully imported
Container autotestme successfully created
must specify at least one container source
Start tests
Run tests for extension Tests-TestLibraries
Creating user manager\containeradministrator as user
Run tests for extension Tests-Bank
  Codeunit 132561 Test Bank Account Tables Success (9.575 seconds)
  Codeunit 134141 ERM Bank Reconciliation Failure (56.9 seconds)
    Testfunction PostPmtReconciliationJournalWhenPostingDateOf2ndLineIsBefore1st Failure (1.334 seconds)
  Codeunit 134231 ERM Bank Account Success (2.194 seconds)
  Codeunit 134252 Match Bank Reconciliation - UT Failure (2.087 seconds)
    Testfunction OneBankEntryMoreRecLinesDateRange Failure (0.17 seconds)
  Codeunit 134253 Match Bank Rec. Scenarios Success (4.291 seconds)
  Codeunit 134259 Bank Pmt. Appl. Algorithm II Success (5.05 seconds)
  Codeunit 134260 Bank Pmt. Appl. Rule UT Success (1.003 seconds)
  Codeunit 134261 Bank Pmt. Appl. Algorithm Failure (96.365 seconds)
    Testfunction TestCustSingleAmountMatchWithAmountPercentageTolerance Failure (0.533 seconds)
    Testfunction TestVendSingleAmountMatchWithAmountPercentageTolerance Failure (0.64 seconds)
  Codeunit 134262 Bank Pmt. Appl. Tolerance Success (188.384 seconds)
  Codeunit 134263 Test Bank Payment Application Failure (15.598 seconds)
    Testfunction BankPaymentApplicationBankReconciliationToBankAccountDiffCurrencies Failure (0.11 seconds)
  Codeunit 134264 Applied Payment Entries Test Success (10.849 seconds)
  Codeunit 134267 Payment Proposal UT Success (22.312 seconds)
  Codeunit 134268 Payment Proposal UT 2 Success (15.496 seconds)
  Codeunit 134269 Matching on Payment Discounts Success (15.837 seconds)
  Codeunit 134271 Payment Recon. E2E Tests Perf. Success (31.836 seconds)
  Codeunit 134500 ERM Cash Manager Success (6.924 seconds)
  Codeunit 134711 Autom. Payment Registration.UX Success (0.987 seconds)
  Codeunit 134712 Autom. Payment Registration.UT Success (0.545 seconds)
Run tests for extension Tests-Cash Flow
  Codeunit 134550 ERM Cash Flow Simplifications Success (9.233 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Skipped
  Codeunit 134553 ERM Cash Flow - Filling II Failure (34.284 seconds)
    Testfunction CustLEWithinDsctPmtTolDsctPmt Failure (0.313 seconds)
    Testfunction VendLEWithinDsctPmtTolDsctPmt Failure (0.303 seconds)
    Testfunction ForecastingFCYAmountConsideringDiscountOnCustLE Failure (0.11 seconds)
    Testfunction ForecastingFCYAmountConsideringDiscountAndCFPmtTermsOnCustLE Failure (0.08 seconds)
    Testfunction ForecastingFCYAmountConsideringDiscountOnVendLE Failure (0.077 seconds)
    Testfunction ForecastingFCYAmountConsideringDiscountAndCFPmtTermsOnVendLE Failure (0.06 seconds)
    Testfunction SalesOrderInFCYWithGroupByDocumentTypeOption Failure (0.033 seconds)
    Testfunction SalesOrderInFCYWithoutGroupByDocumentTypeOption Failure (0.016 seconds)
    Testfunction SalesOrderInFCYWithDiscount Failure (0.077 seconds)
    Testfunction SalesOrderInFCYWithCashFlowPaymentTerms Failure (0.077 seconds)
    Testfunction SalesOrderInFCYWithDiscountAndCashFlowPaymentTerms Failure (0.093 seconds)
    Testfunction PurchaseOrderInFCYWithGroupByDocumentTypeOption Failure (0.013 seconds)
    Testfunction PurchaseOrderInFCYWithoutGroupByDocumentTypeOption Failure (0.016 seconds)
    Testfunction PurchaseOrderInFCYWithDiscount Failure (0.083 seconds)
    Testfunction PurchaseOrderInFCYWithCashFlowPaymentTerms Failure (0.08 seconds)
    Testfunction PurchaseOrderInFCYWithDiscountAndCashFlowPaymentTerms Failure (0.097 seconds)
    Testfunction ServiceOrderInFCYWithGroupByDocumentTypeOption Failure (0.02 seconds)
    Testfunction ServiceOrderInFCYWithoutGroupByDocumentTypeOption Failure (0.016 seconds)
    Testfunction ServiceOrderInFCYWithDiscount Failure (0.08 seconds)
    Testfunction ServiceOrderInFCYWithCashFlowPaymentTerms Failure (0.076 seconds)
    Testfunction ServiceOrderInFCYWithDiscountAndCashFlowPaymentTerms Failure (0.093 seconds)
  Codeunit 134554 ERM Cash Flow - Posting Success (4.187 seconds)
  Codeunit 134555 ERM CF Accounts Success (3.747 seconds)
  Codeunit 134556 ERM CF GL Budget Success (1.774 seconds)
  Codeunit 134557 ERM Cash Flow UnitTests Success (9.43 seconds)
  Codeunit 134558 ERM Cash Flow Pages Failure (42.063 seconds)
    Testfunction ServiceOrdersInCFAvailabilityByPeriodMatrixViewByPeriod Failure (0.864 seconds)
  Codeunit 134559 ERM Cash Flow Navigate Success (12.824 seconds)
  Codeunit 134560 Cash Flow Setup Test Success (1.536 seconds)
  Codeunit 134989 ERM Cash Flow - Reports Success (9.214 seconds)
  Codeunit 135201 Cash Flow Frcst. Handler Tests Success (2.637 seconds)
  Codeunit 135203 CF Frcst. Azure AI Success (6.026 seconds)
  Codeunit 139315 CF Forecast Wizard Tests Success (7.258 seconds)
Run tests for extension Tests-Cost Accounting
  Codeunit 134392 ERM Cost Accounting Rep - Test Success (0.25 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.222 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.219 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.221 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.22 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.22 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.219 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.212 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.219 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.22 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.223 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.219 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.219 seconds)
...
```

As you can see, when the test tool runs into my disabled test codeunit for the first time, it skips it as expected:

```
Run tests for extension Tests-Cash Flow
  Codeunit 134550 ERM Cash Flow Simplifications Success (9.233 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Skipped
```

But when it comes across it for the second time, it keeps rerunning that test codeunit
```
Run tests for extension Tests-Cost Accounting
  Codeunit 134392 ERM Cost Accounting Rep - Test Success (0.25 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.222 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.219 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.221 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.22 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.22 seconds)
  Codeunit 134551 ERM Cash Flow Filling I Success (0.219 seconds)
...
```

I would guess that [this](https://github.com/microsoft/navcontainerhelper/blob/master/AppHandling/PsTestFunctions.ps1#L121,L123) doesn't work in that particular scenario but with little knowledge about the inner workings of the ClientContext, I couldn't figure out how to fix this

**Comment by freddydk** [comment]
This is due to validation errors in the page (which I didn't check for before the next version)
Issue #717 had exactly the same outcome, where it ran the same test again and again (without running it).
I have just checked in a new PSTestFunctions.ps1 in the dev branch (see #717). You can grab this and retry - then it fails with a validation error and you can see the problem.
If this problem then is something we need to fix in the runner, then add it here.

**Comment by freddydk** [comment]
Shipped in 0.6.4.18
